helper/ipsec.c

```diff
@@ -0,0 +1,157 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ * Copyright(C) 2020 Marvell.
+ */
+
+#include <odp/helper/ipsec.h>
+#include <odp/helper/odph_debug.h>
+
+int odph_ipsec_capability_check(odp_bool_t ah,
+				odp_cipher_alg_t cipher_alg,
+				uint32_t cipher_key_len,
+				odp_auth_alg_t auth_alg,
+				uint32_t auth_key_len)
+{
+	odp_ipsec_capability_t capa;
+	odp_ipsec_cipher_capability_t cipher_capa[MAX_ALG_CAPA];
+	odp_ipsec_auth_capability_t   auth_capa[MAX_ALG_CAPA];
+	int i, num;
+	odp_bool_t found;
+
+	if (odp_ipsec_capability(&capa) < 0)
```

Comment:
I do not this this AH capa checks belongs here. There are plenty of various capability flags in the capability structure and it does not make sense to check them all here (requiring a huge number of parameters too). An application can as easily check the relevant capability flags itself.

The crypto algorithm check is the useful part that improves code reuse and that is what this function should concentrate on. The name should then be changed accordingly (odp_ipsec_alg_check?).

Maybe the function should take odp_ipsec_capability_t as a parameter?

Reply:
Ack.
Updated the helper function to just check algorithm specific capabilities.
Also added odp_ipsec_capability_t as a new parameter that needs to be passed to the function.